table/block_based/block_based_table_reader.cc

```diff
@@ -1187,8 +1188,10 @@ Status BlockBasedTable::Open(
   rep->hash_index_allow_collision = table_options.hash_index_allow_collision;
   // We need to wrap data with internal_prefix_transform to make sure it can
   // handle prefix correctly.
-  rep->internal_prefix_transform.reset(
-      new InternalKeySliceTransform(prefix_extractor));
+  if (prefix_extractor != nullptr) {
```

Comment:
The builder has a variable with the same name but different property that it is never nullptr. If we are going to keep the change, can we add inline comments above internal_prefix_transform clarifying that when it is expected to be nullptr?

Reply:
Sure. Note that, if InternalKeySliceTransform is built using a null inner extractor, it's not usable anyway even if it is not null.